actors/builtin/miner/miner_test.go

```diff
@@ -1114,146 +1165,98 @@ func TestProvingPeriodCron(t *testing.T) {
 		WithBalance(bigBalance, big.Zero())
 
 	t.Run("empty periods", func(t *testing.T) {
-		t.Skip("Disabled in miner state refactor #648, restore soon")
 		rt := builder.Build(t)
 		actor.constructAndVerify(rt)
 		st := getState(rt)
 		assert.Equal(t, periodOffset, st.ProvingPeriodStart)
 
-		// First cron invocation just before the first proving period starts.
-		rt.SetEpoch(periodOffset - 1)
+		// crons before proving period do nothing
 		secondCronEpoch := periodOffset + miner.WPoStProvingPeriod - 1
-		actor.onDeadlineCron(rt, &cronConfig{
-			expectedEntrollment: secondCronEpoch,
-		})
+		dlinfo := actor.deadline(rt)
+		for dlinfo.Close < secondCronEpoch {
+			dlinfo = advanceDeadline(rt, actor, &cronConfig{})
+		}
+
 		// The proving period start isn't changed, because the period hadn't started yet.
 		st = getState(rt)
 		assert.Equal(t, periodOffset, st.ProvingPeriodStart)
 
-		rt.SetEpoch(secondCronEpoch)
-		actor.onDeadlineCron(rt, &cronConfig{
-			expectedEntrollment: periodOffset + 2*miner.WPoStProvingPeriod - 1,
-		})
-		// Proving period moves forward
+		// next cron moves proving period forward
+		advanceDeadline(rt, actor, &cronConfig{})
 		st = getState(rt)
 		assert.Equal(t, periodOffset+miner.WPoStProvingPeriod, st.ProvingPeriodStart)
 	})
 
-	t.Run("first period gets randomness from previous epoch", func(t *testing.T) {
```

Comment:
Is this functionality completely removed after the refactor?

Reply:
It is very unclear what this is supposed to be testing. The code itself just appears to be going though a couple proving periods and occasionally setting the now non-existant `NewSectors`. The test description doesn't really make literal sense even when "first period" is well defined. I confirmed that every deadline in a proving period uses the same formula for `deadlineInfo.Challenge`, so maybe whatever behavior this is testing has gone away.